hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -286,6 +286,12 @@ private void handleUserTransaction(
             @NonNull final ConsensusEvent platformEvent,
             @NonNull final NodeInfo creator,
             @NonNull final ConsensusTransaction platformTxn) {
+        // Determine if this is the first transaction after startup. This needs to be determined BEFORE starting the
+        // user transaction
+        final var firstConsTimeOfLastBlock = blockRecordManager.firstConsTimeOfLastBlock();
+        final var isFirstTransaction =
+                firstConsTimeOfLastBlock != null && !firstConsTimeOfLastBlock.isAfter(Instant.EPOCH);
+
```

Comment:
How can we decide its startup based on `firstConsTimeOfLastBlock()` ? 🤔 @tinker-michaelj  thoughts ?

Reply:
I wrote this, but can't remember why I didn't use the last handled transaction timestamp. I'll run some tests to see if I can remember.

Reply:
Just a mistake on my part. I pushed a change that uses the last handled txn timestamp instead. Good catch!